client/landing/stepper/declarative-flow/internals/steps-repository/unified-domains/index.tsx

```diff
@@ -93,10 +93,9 @@ export default function DomainsStep( props: StepProps ) {
 				}
 				submitSignupStep={ ( state: ProvidedDependencies ) => {
 					setStepState( { ...stepState, ...state } );
+					props.navigation.submit?.( state );
 				} }
-				goToNextStep={ ( state: ProvidedDependencies ) =>
-					props.navigation.submit?.( { ...stepState, ...state } )
-				}
+				goToNextStep={ () => {} }
```

Comment:
This prop is marked as [required](https://github.com/Automattic/wp-calypso/blob/fix/stepper-domain-submit/client/signup/steps/domains/index.jsx#L94). Maybe we could make it not required and get rid of passing it with an empty function?